test/v2/USTB.t.sol

```diff
@@ -0,0 +1,1563 @@
+pragma solidity ^0.8.20;
+
+import "forge-std/StdUtils.sol";
+import {Test} from "forge-std/Test.sol";
+import {console} from "forge-std/console.sol";
+import {PausableUpgradeable} from "openzeppelin-contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import "openzeppelin-contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "openzeppelin-contracts/proxy/transparent/ProxyAdmin.sol";
+import {SuperstateToken as SuperstateTokenV1} from "src/SuperstateToken.sol";
+import {SuperstateToken} from "src/v2/SuperstateToken.sol";
+import {USTB as USTBV1} from "src/USTB.sol";
+import {USTB} from "src/v2/USTB.sol";
+import {AllowList} from "src/AllowList.sol";
+import "test/AllowListV2.sol";
+import "test/USTBV2.sol";
+
+contract USTBv2Test is Test {
```

Comment:
Tests are mirrored like this so that I can upgrade the proxy from v1 to v2 in the `setup()`.

This gives some additional confidence w/ stress testing the storage layout. However, further tests will be written that fork the test of mainnet

Reply:
Can you make the test functions reusable instead of copy paste?